### ISSUE TITLE
`BatchGradientVerification` check always returns `False` 

### ISSUE THREAD

**Issue by armanal** [awaelchli/pytorch-lightning-snippets#3]
Hi

it seems that `BatchGradientVerification.check` allways returns `False` when there are `BatchNorm` layers 
present in the network architecture.

to reproduce

```
import torch
from verification.batch_gradient import BatchGradientVerification

model = torch.nn.Sequential(
    torch.nn.Conv1d(
        1,
        2,
        5,
        bias=False,
    ),
    torch.nn.BatchNorm1d(2),
    torch.nn.ReLU(),
)
data = torch.rand(4, 1, 100)

verification = BatchGradientVerification(model)
valid = verification.check(input_array=data, sample_idx=2)
print("Mixes data across batch dimention." if not valid else "OK.")
```
